Sources/StreamChat/Controllers/SearchControllers/UserSearchController/UserSearchController.swift

```diff
@@ -195,16 +96,129 @@ public class ChatUserSearchController: DataController, DelegateCallable, DataSto
         limit: Int = 25,
         completion: ((Error?) -> Void)? = nil
     ) {
-        guard let lastQuery = lastQuery else {
+        guard let lastQuery = query else {
             completion?(ClientError("You should make a search before calling for next page."))
             return
         }
         
         var updatedQuery = lastQuery
-        updatedQuery.pagination = Pagination(pageSize: limit, offset: users.count)
-        userQueryUpdater.update(userListQuery: updatedQuery) { error in
-            self.callback { completion?(error) }
+        updatedQuery.pagination = Pagination(pageSize: limit, offset: userArray.count)
+        
+        fetch(updatedQuery, completion: completion)
+    }
+}
+
+private extension ChatUserSearchController {
+    /// Fetches the given query from the API, saves the loaded page to the database, updates the list of users and notifies the delegate.
+    ///
+    /// - Parameters:
+    ///   - query: The query to fetch.
+    ///   - completion: The completion that is triggered when the query is processed.
+    func fetch(_ query: UserListQuery, completion: ((Error?) -> Void)? = nil) {
+        // TODO: Remove with the next major
```

Comment:
@polqf in the user's search controller we stopped persisting the results in the database so it does nothing with local data anymore therefore the `initialized -> localDataFetched` state transition no longer makes sense. 

However, customers may have a logic that involves checking `localDataFetched` (in DemoApp we do as well). So to do not break it the first part of `initialized -> localDataFetched -> remoteDataFetched` transition is done manually when one of the `fetch` methods is invoked.

With the next major we can stop doing that since it will be expected by customers that some parts can start behaving differently.